code/modules/mob/living/silicon/ai/freelook/eye.dm

```diff
@@ -188,6 +188,7 @@
 
 	if(isturf(loc) && (QDELETED(eyeobj) || !eyeobj.loc))
 		to_chat(src, "ERROR: Eyeobj not found. Creating new eye...")
+		QDEL_NULL(eyeobj)
```

Comment:
Could you put a stack trace here if this is in fact an error case

Reply:
It wouldn't be useful, it wouldn't give any information about what actually qdel'd it.

The stack trace would only give the one call site, which is the AI clicking jump to eye.

Reply:
Yes but it logs that it happened in the first place, which I think is useful

Reply:
Also QDEL_NULL will just runtime if this is already qdeleted, you sure you shouldn't add more exact checks here?

Reply:
> Also QDEL_NULL will just runtime if this is already qdeleted

No it won't, what are you talking about?

Reply:
The logging thing is a fair point though

```suggestion
		stack_trace("AI eye object wasn't found! Location: [loc] / Eyeobj: [eyeobj] / QDELETED: [QDELETED(eyeobj)] / Eye loc: [eyeobj.loc]")
		QDEL_NULL(eyeobj)
```

Reply:
Ah you're right, that check is more particular then I remembered